### PR TITLE
SemanticKernel: Correcting non-standard way of working with PromptExecutionSettings

### PR DESCRIPTION
**Pull request by zsogitbe**
The extension of PromptExecutionSettings is not only for ChatCompletion, but also for text completion and text embedding.

### CODE REVIEW

**Comment by zsogitbe** [comment]
Sorry for the list of commits. Until one is not merged it keeps listing all former ones... only the last commit is active here: 
[Correcting non-standard way of working with PromptExecutionSettings](https://github.com/SciSharp/LLamaSharp/pull/689/commits/ab8dd0dfc7604249b70cf73334245e953377949f) 
[ab8dd0d](https://github.com/SciSharp/LLamaSharp/pull/689/commits/ab8dd0dfc7604249b70cf73334245e953377949f)

**Comment by martindevans** [comment]
If you work in a separate branch in your fork and then PR just that branch you won't get a messy commit log like that. For this one, we'll just make sure to squash-and-merge so it all becomes on commit :)

File: LLama.SemanticKernel/LLamaSharpPromptExecutionSettings.cs [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+using LLamaSharp.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 
-namespace LLamaSharp.SemanticKernel.ChatCompletion;
+namespace LLamaSharp.SemanticKernel;
 
-public class ChatRequestSettings : PromptExecutionSettings
+public class LLamaSharpPromptExecutionSettings : PromptExecutionSettings
```
Comment (AsakusaRinne) [inline LLama.SemanticKernel/LLamaSharpPromptExecutionSettings.cs:19]: This is a reasonable renaming but introduces a break change. Could you please keep the `ChatRequestSettings` but mark it as `Obsolete`? It'll be better to completely remove it in the next 1~2 releases.

**Review by AsakusaRinne — CHANGES_REQUESTED** [review]

File: LLama.SemanticKernel/LLamaSharpPromptExecutionSettings.cs (hunk 1 above)
Comment (zsogitbe) [reply LLama.SemanticKernel/LLamaSharpPromptExecutionSettings.cs:19]: OK!

**Review by AsakusaRinne — APPROVED** [review]

**Pull request merged**